CRM/Financial/Form/FinancialAccount.php

```diff
@@ -143,7 +143,7 @@ public static function formRule($values, $files, $self) {
       if ($values['financial_account_type_id'] != $financialAccountTypeId) {
         $errorMsg['financial_account_type_id'] = ts('Taxable accounts should have Financial Account Type set to Liability.');
       }
-      if (CRM_Utils_Array::value('tax_rate', $values) == NULL) {
+      if (!isset($values['tax_rate'])) {
         $errorMsg['tax_rate'] = ts('Please enter value for tax rate');
```

Comment:
I started at the bottom and have looked upward to about here. Strictly speaking these two lines aren't identical, it just happens to work out because of the context, and actually the original is probably flawed. If $values['tax_rate'] is integer 0 instead of string 0 then the original is true but the replacement is false. However because it's a form value it's a string, and you do want it to evaluate to false here because 0 is an allowed value. So both statements work in context here but similar ones elsewhere might need to take into account whether it could be integer 0 or string 0.